examples/enum/main.go

```diff
@@ -0,0 +1,111 @@
+package enum
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/podhmo/goat"
+	"github.com/podhmo/goat/examples/enum/customtypes"
+)
+
+//go:generate goat emit -run Run -initializer NewOptions main.go
+
+// MyLocalEnum is a locally defined enum type.
+type MyLocalEnum string
+
+const (
+	// LocalA is a possible value for MyLocalEnum.
+	LocalA MyLocalEnum = "local-a"
+	// LocalB is a possible value for MyLocalEnum.
+	LocalB MyLocalEnum = "local-b"
+)
+
+// GetLocalEnumsAsStrings returns the string representations of MyLocalEnum options.
+func GetLocalEnumsAsStrings() []string {
+	return []string{string(LocalA), string(LocalB)}
+}
+
+// Options defines the command line options for the enum example.
+type Options struct {
+	// LocalEnumField demonstrates a locally defined enum.
+	LocalEnumField MyLocalEnum `env:"ENUM_LOCAL_ENUM"`
+
+	// ImportedEnumField demonstrates an enum imported from another package.
+	ImportedEnumField customtypes.MyCustomEnum `env:"ENUM_IMPORTED_ENUM"`
+
+	// OptionalImportedEnumField demonstrates an optional enum (pointer type)
+	// imported from another package.
+	OptionalImportedEnumField *customtypes.MyCustomEnum `env:"ENUM_OPTIONAL_IMPORTED_ENUM"`
+}
+
+// NewOptions initializes Options with default values and enum constraints.
+func NewOptions() *Options {
+	return &Options{
+		LocalEnumField: goat.Default(LocalA, goat.Enum(GetLocalEnumsAsStrings())),
+		ImportedEnumField: goat.Default(customtypes.OptionX, goat.Enum(customtypes.GetCustomEnumOptionsAsStrings())),
```

Comment:
```suggestion
// TODO: Since it's a generic function, there should be no need to cast it to a string type.If there are reasons why it is not possible, I want to resolve them.
		ImportedEnumField: goat.Default(customtypes.OptionX, goat.Enum(customtypes.GetCustomEnumOptionsAsStrings())),
```